app/assets/javascripts/app/calendar.js.coffee

```diff
@@ -0,0 +1,73 @@
+class window.FullCalendarConfig
+  constructor: (@$element) ->
+
+  init: (options) ->
+    @$element.fullCalendar($.extend(@options(), options))
+
+  options: ->
+    options = @baseOptions()
+    if window.minTime?
+      options.minTime = window.minTime
+    if window.maxTime?
+      options.maxTime = window.maxTime
+      options.height = 42 * (maxTime - minTime) + 75
+    if window.initialDate
+      d = Date.parse(initialDate)
+      $.extend(options,
+        year: d.getFullYear()
+        month: d.getMonth()
+        date: d.getDate())
+    options
+
+  baseOptions: ->
+    editable: false
+    defaultView: "agendaWeek"
+    allDaySlot: false
+    events: events_path
+    loading: (isLoading, view) =>
+      @toggleOverlay(isLoading)
+
+    eventAfterRender: @buildTooltip
+    eventAfterAllRender: @toggleNextPrev
+
+  toggleOverlay: (isLoading) ->
+    if isLoading
+      $("#overlay").addClass("on").removeClass("off")
+    else
+      $("#overlay").addClass("off").removeClass("on")
+
+  toggleNextPrev: (view) ->
+    # window.minDate/maxDate are strings formatted like 20170714
+    try
+      startDate = $.fullCalendar.formatDate(view.start, "yyyyMMdd")
+      endDate   = $.fullCalendar.formatDate(view.end, "yyyyMMdd")
+
```

Comment:
I didn't really change this piece of the code, other than to add a comment about why it's this way, but pulling it into a method is probably a good idea.